pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -308,6 +309,12 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ct
 			return
 		}
 
+		// If the namespace is deleted during this run of the reconciliation loop, then there's
+		// nothing to update.
+		if apierrors.IsNotFound(updateErr) {
+			return
+		}
```

Comment:
How are we sure this not found relates to a deleted namespace?

Reply:
this is different than the replicaset code you've set - it is for an updateError - it means we were able to get an ExternalSecret, and while performing an update, the ExternalSecret was deleted (possibly due to namespace failure).